packages/utils/src/carequality/e2e-validation.ts

```diff
@@ -0,0 +1,181 @@
+      state: USState.AL,
+      city: "Helena",
+      country: "USA",
+    },
+  ],
+};
+
+async function main() {
+  let patientId = "";
+
+  try {
+    console.log("Creating patient...");
+    const newPatient = await metriportAPI.createPatient(testPatient, facilityId);
+
+    patientId = newPatient.id;
+
+    console.log("Created patient:", newPatient);
+
+    await sleep(PATIENT_SLEEP.asMilliseconds());
+
+    const cqPatientDataQuery = `SELECT * FROM ${cqPatientDataTableName} WHERE id = '${newPatient.id}'`;
+    let cqPatientDataResults = await sequelize.query(cqPatientDataQuery, {
+      type: QueryTypes.SELECT,
+    });
+
+    while (cqPatientDataResults.length === 0) {
+      await sleep(PATIENT_SLEEP.asMilliseconds());
+      cqPatientDataResults = await sequelize.query(cqPatientDataQuery, {
+        type: QueryTypes.SELECT,
+      });
+    }
+
+    const cqPatientData = cqPatientDataResults[0] as any; // eslint-disable-line @typescript-eslint/no-explicit-any
+    console.log("cqPatientData:", cqPatientData);
+    const links = cqPatientData.data.links;
+
+    if (cqPatientDataResults.length === 0) {
+      throw new Error(`Patient not found in ${cqPatientDataTableName}`);
+    }
+
+    console.log("Patient links found:", cqPatientDataResults[0]);
+
+    const patientDiscoveryResults = await queryDbResults(
+      patientDiscoveryResultTableName,
+      newPatient.id
+    );
+
+    if (patientDiscoveryResults.length === 0) {
+      throw new Error(`Patient not found in ${patientDiscoveryResultTableName}`);
+    }
+
+    console.log("Patient discovery results found:", patientDiscoveryResults);
+
+    console.log("Start document query");
+
+    const docQuery = await metriportAPI.startDocumentQuery(newPatient.id, facilityId);
+
+    console.log("Document query started:", docQuery);
+
+    const documentQueryResults = await pollOutboundDocQueryResults({
+      patientId: newPatient.id,
+      cxId,
+      dbCreds: sqlDBCreds,
+      requestId: docQuery.requestId ?? "",
+      numOfGateways: links.length,
+    });
+
+    if (documentQueryResults.length === 0) {
+      throw new Error("No document query results found");
+    }
+
+    console.log("Document query results found:", documentQueryResults);
+
+    const documentRetrievalResults = await pollOutboundDocRetrievalResults({
+      patientId: newPatient.id,
+      cxId,
+      dbCreds: sqlDBCreds,
+      requestId: docQuery.requestId ?? "",
+      numOfGateways: links.length,
+    });
+
+    if (documentRetrievalResults.length === 0) {
+      throw new Error("No document retrieval results found");
+    }
+
+    console.log("Document retrieval results found:", documentRetrievalResults);
+
+    await sleep(DOC_STATUS_SLEEP.asMilliseconds());
+
+    let docQueryStatus = await metriportAPI.getDocumentQueryStatus(newPatient.id);
+
+    while (
+      docQueryStatus.download?.status === "processing" &&
+      docQueryStatus.convert?.status === "processing"
```

Comment:
Do we want to continue if only one is not `processing`?

Also, we're not entering the loop if the `download` or `convert` are not set.

Reply:
Updated